src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,206 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+public class Duke {
+    static String horizontalLine = "--------------------------------------------------";
+    static int thingsCounted = 0;
+    //static Task[] tasks = new Task[100];
+    public static ArrayList<Task> taskArray = new ArrayList<>();
+    public static final String FILE_PATH = "duke.txt";
+
+    public static void writeToFile(String FILE_PATH){
+        try {
+            File f = new File(FILE_PATH);
+            FileWriter fw = new FileWriter(FILE_PATH);
+            for(Task task: taskArray){
+                fw.write(task.printIntoFile() + "\n");
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Error writing file");
+        }
+    }
+
+    public static void readFromFile(String FILE_PATH){
+        try {
+            File f = new File(FILE_PATH);
+            f.createNewFile();
+            Scanner sc = new Scanner(f);
+            Task task;
+            while(sc.hasNext()){
+                String[] taskInFile = sc.nextLine().split("\\|");
+                if(taskInFile[0].equals("T")){
+                    task = new Todo(taskInFile[2]);
+                }else if(taskInFile[0].equals("D")){
+                    task = new Deadline(taskInFile[2],taskInFile[3]);
+                }else{
+                    task = new Event(taskInFile[2], taskInFile[3]);
+                }
+                thingsCounted++;
+                if(taskInFile[1].equals("true")){
+                    task.taskDone();
+                }
+                taskArray.add(task);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Error reading file");
+        }catch(IOException e){
+            e.printStackTrace();
+        }
+    }
+
+    public static void storeTextAndList() throws DukeException {
+        String command;
+        Scanner in = new Scanner(System.in);
+        command = in.nextLine();
+        while (!command.equals("bye")) {
+            System.out.println(horizontalLine);
+            try {
+                if (command.equals("list")) {
+                    listTask();
+                } else if (command.contains("done")) {
+                    int index = Integer.parseInt(command.substring(5));
```

Comment:
Might want to consider extracting this into a function so that SLAP is applied to all, much like how the other commands are doing it.